src/station/station_info.rs

```diff
@@ -0,0 +1,946 @@
+const NL80211_STA_INFO_AIRTIME_WEIGHT: u16 = 40;
+const NL80211_STA_INFO_AIRTIME_LINK_METRIC: u16 = 41;
+const NL80211_STA_INFO_ASSOC_AT_BOOTTIME: u16 = 42;
+const NL80211_STA_INFO_CONNECTED_TO_AS: u16 = 43;
+
+/// Station information
+///
+/// These attribute types are used with [`Nl80211Attr::Nl80211StationInfo`]
+/// when getting information about a station.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Nl80211StationInfo {
+    /// Time since last activity (msecs)
+    InactiveTime(u32),
+    /// Total transmitted bytes (MPDU length)
+    TxBytes(u32),
+    /// Total received bytes (MPDU length)
+    RxBytes(u32),
+    /// Total transmitted bytes (MPDU length)
+    TxBytes64(u64),
+    /// Total received bytes (MPDU length)
+    RxBytes64(u64),
+    /// Signal strength of last received PPDU (dBm)
+    Signal(u8),
+    /// Signal strength average (u8, dBm)
+    SignalAvg(u8),
+    /// Current unicast transmission rate, nested attribute containing info as
+    /// possible, see [`Nl80211RateInfo`]
+    TxBitrate(Vec<Nl80211RateInfo>),
+    ///  Last unicast data frame receive rate, nested attribute, like
+    /// [`NL80211_STA_INFO_TX_BITRATE`].
+    RxBitrate(Vec<Nl80211RateInfo>),
+    /// Total transmitted packets (MSDUs and MMPDUs)
+    TxPackets(u32),
+    /// Total received packets (MSDUs and MMPDUs)
+    RxPackets(u32),
+    /// Total number of received packets (MPDUs)
+    RxMpdus(u32),
+    /// Total retries (MPDUs)
+    TxRetries(u32),
+    /// Total failed packets (MPDUs) (u32, to this station)
+    TxFailed(u32),
+    /// Transmitted packets dropped for unspecified reasons
+    RxDropMisc(u64),
+    /// Total number of packets (MPDUs) received with an FCS error . This count
+    /// may not include some packets with an FCS error due to TA corruption.
+    /// Hence this counter might not be fully accurate.
+    FcsErrorCount(u32),
+    /// The station's mesh LLID
+    Llid(u16),
+    /// The station's mesh PLID
+    Plid(u16),
+    /// Peer link state for the station (see [`Nl80211PeerLinkState`])
+    PeerLinkState(Nl80211PeerLinkState),
+    /// Current station's view of BSS, nested attribute containing info as
+    /// possible, see [`Nl80211StationBssParam`]
+    BssParam(Vec<Nl80211StationBssParam>),
+    /// Time since the station is last connected
+    ConnectedTime(u32),
+    /// Contains a [Nl80211StationFlagUpdate]
+    StationFlags(Nl80211StationFlagUpdate),
+    /// Timing offset with respect to this station
+    TimingOffset(i64),
+    /// Local mesh staion link-specific power mode
+    LocalPowerMode(Nl80211MeshPowerMode),
+    /// Peer mesh staion link-specific power mode
+    PeerPowerMode(Nl80211MeshPowerMode),
+    /// Neighbor mesh station power save mode towards non-peer station
+    NonPeerPowerMode(Nl80211MeshPowerMode),
+    /// Per-chain signal strength of last PPDU. Contains a nested array of
+    /// signal strength attributes (dBm)
+    ChainSignal(Vec<u8>),
```

Comment:
Can we somehow abstract this into user friendly struct stead of `[u8]`?

Reply:
Not sure how to wrap this in other than a single element tuple.

Reply:
You are right, the kernel is just exposing it as Vec<u8>.

```c
	for (i = 0; i < IEEE80211_MAX_CHAINS; i++) {
		if (!(mask & BIT(i)))
			continue;

		if (nla_put_u8(msg, i, signal[i]))
			return false;
	}
```